File: tests/minimum_size_subarray_sum_tests.cpp
#include "pch.h"
#include "minimum_size_subarray_sum.h"
using namespace std;

namespace LeetcodeSolutions {
TEST(MinimSubarraySum, ShouldReturn0IfNumsArrayIsEmpty) {
	vector<int> nums = {};
	int target = 1;
	int expected = 0;
	int actual = Solution().minSubArrayLen(target, nums);
	EXPECT_EQ(expected, actual);
}

TEST(MinimSubarraySum, ShouldReturn1IfNumsArrayContainsNumbersWhichSumUpToTarget) {
	vector<int> nums = { 1, 2, 3, 4, 5 };
	int target = 5;
	int expected = 1;
	int actual = Solution().minSubArrayLen(target, nums);
	EXPECT_EQ(expected, actual);
}
}

File: Tests/MinimumSizeSubarraySumTests.cs
using Xunit;
using Solutions.Minimumsizesubarraysum;

namespace Tests;

public class MinimumSizeSubarraySumTests
{
    [Theory]
    [InlineData(new [] {2,3,1,2,4,3}, 7, 2)]
    [InlineData(new [] {1,4,4}, 4, 1)]
    [InlineData(new [] {1,1,1,1,1,1,1,1}, 11, 0)]
    public void TestSolution(int[] nums, int target, int expectedLength)
    {
        var solution = new Solution();
        Assert.Equal(expectedLength, solution.MinSubarrayLen(target, nums));
    }
}
